### PR TITLE
eriks messings with Jump

### PR DESCRIPTION
**Pull request by nandemoguy**
added non colidable "foreground" stuff and only top of one-way stack now colidable.

### CODE REVIEW

File: jump-core/src/main/java/com/bitdecay/jump/level/builder/LevelBuilder.java [hunk 1]
```diff
@@ -274,11 +284,11 @@ public void updateOwnNeighbors(int x, int y) {
 		}
 		// check left
 		if (ArrayUtilities.onGrid(grid, x - 1, y) && grid[x - 1][y] != null) {
-			if (grid[x-1][y].oneway) {
-				grid[x][y].collideNValue &= Direction.NOT_LEFT;
-			} else {
+			//if (grid[x-1][y].oneway) {
+			//	grid[x][y].collideNValue &= Direction.NOT_LEFT;
+			//} else {
```
Comment (MondayHopscotch) [inline jump-core/src/main/java/com/bitdecay/jump/level/builder/LevelBuilder.java]: I think we still want these. Commenting these out will result in bugs when one-way platforms butt up against regular terrain.

Comment (nandemoguy) [inline jump-core/src/main/java/com/bitdecay/jump/level/builder/LevelBuilder.java]: Ah indeed. Forgot to uncomment them.

**Comment by MondayHopscotch** [comment]
I like these changes, but the more I think about it -- I feel they are more game-specific than what I want to keep in the engine. So where I think this code belongs is in the LevelSegmentAggregator class inside our Dendrite code.

In there, there is a method called `updateAllNeighborRenderValues` that we are already stepping through to do some custom tile manipulation. I think that would be the perfect place for this code to live. (You'll notice it looks nearly identical to where you were working... because it mostly is. But it allows us to do some custom logic.)

So I'm not going to merge this PR. If you have questions, I should have some time tomorrow evening to chat about it.

**Comment by nandemoguy** [comment]
No sweat man. I'll look in that other place tomorrow sometime. Was mostly
just looking to see if I could figure out how to do some stuff as opposed
to making any real lasting changes to the overall program.
A fun challenge to myself.

**Comment by MondayHopscotch** [comment]
The changes will literally be the same code -- just in a different place. So I'm not discrediting what you did, just where you did it :) (I would have been shocked if you put it where I was talking about on your first time through the code)

**Comment by nandemoguy** [comment]
Oh for sure. I'm not at all worried about it. I'll look again tomorrow and
do some more playing. And maybe in the right places hah.

**Comment by nandemoguy** [comment]
Hey this update own neighbor only seems to affect the render but not the
colisions is that right?

**Comment by MondayHopscotch** [comment]
Currently, that is all it does -- but you are free to change it to do what
you want.

**Comment by MondayHopscotch** [comment]
Foreground and Background elements have been implemented as part of #203. Submit new issues for any other desired behavior.

**Comment by nandemoguy** [comment]
Which branch do I need to pull? Just jump?

**Pull request closed without merging**